oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/index/ElasticIndexWriter.java

```diff
@@ -204,11 +204,35 @@ private void provisionIndex() throws IOException {
             if (ese.status() == 400 && ese.getMessage().contains("resource_already_exists_exception")) {
                 LOG.warn("Index {} already exists. Ignoring error", indexName);
             } else {
+                LOG.warn("Failed to create index {}", indexName, ese);
+                StringBuilder sb = new StringBuilder();
+                int old = JsonpUtils.maxToStringLength();
+                try {
+                    JsonpUtils.maxToStringLength(16_000_000);
+                    JsonpUtils.toString(request, sb);
+                    String[] array = splitLargeString(sb.toString(), 1024);
+                    for (int i = 0; i < array.length; i++) {
+                        LOG.warn("request chunk[{}] = {}", i, array[i]);
+                    }
+                } finally {
+                    JsonpUtils.maxToStringLength(old);
+                }
```

Comment:
This may print 16MB worth of data to the logs, which would likely overload the logger and any log management systems that may be capturing the logs, which could lead to us losing other logs which are more important. And all this data would likely not be useful. I suggest that we truncate the string to a reasonable size, like 1K or so.

Reply:
And allocating 16MB of memory just to create a log message also sounds risky, as it can create memory pressure. For instance, if there are many writers, all hitting the same error, they would all in a short period of time allocate up to 32MB of memory (16MB for the full Json representation, then 16MB for the array split).